### DISCUSSION TITLE
q-input with date picker using popup proxy

### DISCUSSION THREAD

**Discussion by scp-nm** [quasarframework/quasar#9076] [General - Components / Directives / etc] [answered]
Hi all,

I'm trying to replicate what I can do using Vuetify (https://codepen.io/scp-nm/pen/ZELZGaN) where I have an input field that triggers a date picker.

This is my attempt to doing this, but as you'll see it's not looking great - https://codepen.io/scp-nm/pen/qBRwbgz

Where am I going wrong?

Thanks in advance.

**Comment by metalsadman** [comment, marked as answer]
try this https://codepen.io/metalsadman/pen/ExZJrYr?editors=1010.

**Reply by scp-nm** [reply]
Thanks for that @metalsadman - still not quite as elegant as the Vuetify version though.

**Reply by metalsadman** [reply]
different api's, there are strengths and what not in both implementations, It's you who weighs which one you pick. further enhancement lies in the community making PRs, issuing bugs and or feature request.

**Reply by scp-nm** [reply]
Completely agree, and just about every framework I look at has certain cons and some pros, so it will be trying to weigh these up with how we've created the UI. Given that we've got a SaaS platform that uses quite a few date inputs ensuring this is as user friend as possible is a pretty big con at the moment. I'll keep playing with Quasar, as there's a lot I do like about it over Vuetify.

**Reply by h2jose** [reply]
Hi, i amtrying run with composition api, i can get working fine the close popup, i have tried these:

`<q-date v-model="editedItem.date" mask="DD/MM/YYYY" @input="() => $refs.qDateProxy.hide()">`
i get error: 'Property 'hide' does not exist on type 'unknown'(Vetur)

After
`<q-date v-model="editedItem.date" mask="DD/MM/YYYY" @input="$ref.qDateProxy.hide()">`
same error: 'Property 'hide' does not exist on type 'unknown'(Vetur)

after i updated:
`<q-date v-model="date" mask="DD/MM/YYYY" @input="() => qDateProxy.hide()">` 
i got not error but Not working

Who can i do to the select and close works fine in composition api?
Thanks

**Reply by metalsadman** [reply]
@h2jose should be `@update:model` instead of @input and disregard that vetur warning or turn off that rule check (not sure if at eslintrc.js or your ide's settings.json, anyway I use volar now in lieu of vetur).

**Reply by h2jose** [reply]
@metalsadman thanks for the quick response. But not working, the q-popup-proxy not hidde when i click on a day, here my complete code:
`
<q-input dark filled v-model="date">

	<template v-slot:append>

		<q-icon name="event" class="cursor-pointer">
			<q-popup-proxy ref="qDateProxy" transition-show="scale" transition-hide="scale">
				<q-date v-model="date" mask="DD/MM/YYYY" @update:model="$refs.qDateProxy.hide()">
					<div class="row items-center justify-end">
						<q-btn v-close-popup label="Close" color="primary" flat />
					</div>
				</q-date>
			</q-popup-proxy>
		</q-icon>
	</template>
</q-input>
`
after i tried:
`<q-date v-model="date" mask="DD/MM/YYYY" @update:model="qDateProxy.hide()">`
and the last attempt
`<q-date v-model="date" mask="DD/MM/YYYY" @update:model="$ref.qDateProxy.hide()">`


My setup:
`
setup () {

    const date = ref(new Date().toISOString().substr(0, 10))
    const  qDateProxy = ref(null)
	return {
            date,
           qDateProxy,
           save,
	}
}
`
thanks

**Comment by h2jose** [comment]
@metalsadman Your help is greatly appreciated! , i found the solution, the correct directive is:
`
<q-date

   **@update:model-value="qDateProxy.hide()"**
   ...
/q-date>
`
The **$refs** in composition api is not neccesary, in setup():

`

...
const qDateProxy = ref(null)
....

return { 
 ...
qDateProxy,
...
`
